### ISSUE TITLE
libxfce4ui 4.18.4 - User name still / already typed-in after resume from standby

### ISSUE THREAD

**Issue by gc-user** [void-linux/void-packages#44551]
### Is this a new report?

Yes

### System Info

Void 6.1.34_1 x86_64

### Package(s) Affected

libxfce4ui-4.18.4

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

No.

### Expected behaviour

If
greeter-hide-users=true
in /etc/lightdm/lightdm.conf.d/50-lightdm.conf, the user name (short form) should have to be entered into a text box (instead of a drop down menu with all available user names (long form)) and that text box should be empty after boot or resume from standby.

### Actual behaviour

After updating libxfce4ui to v4.18.4 (and reboot), the user name (short form) is still / already typed into the text box after resume from standby.

### Steps to reproduce

1. Log in.
2. Go to standby
3. Resume
4. The user name is still / already typed into the text box for the user name.

"Fix":
1. Log in.
2. Downgrade libxfce4ui to v4.18.3
3. Reboot
4. Log in
5. Go to standby
6. Resume
7. The text box for the user name is empty - as it should be.

**Comment by CabrioletDiskette** [comment]
I'm confused, is this a lightdm issue or a xfce4 issue? If you just put your computer to suspend I believe it goes to the xfce4 locker right? In that case why does the lightdm conf have anything to do with it? If it is a xfce4 locker issue then double check that your computer is going to suspend. I have xfce4+lxdm and when I suspend I get xfce4 locker(which has the username at the top) but when I hibernate or hybrid sleep I obliviously end up at lxdm. Thanks in advance for clarifying.

**Comment by gc-user** [comment]
Well, I'm just reporting my findings.

My first suspects would have been either lightdm, lightdm-gtk3-greeter or light-locker. But none of them had any recent updates. I first tried downgrading xfce4-session, but that didn't change anything, so I checked which other package that was updated around the time the bug appeared. So I tried xfce4ui, which I never noticed existed before.

It might be that one of the light* packages is incompatible with the new xfce4ui package and thus the error occurs, and nothing is really wrong with xfce4ui. But for me and for now, it doesn't matter which package ultimately is the culprit, downgrading xfce4ui made the error disappear from my system.

I mentioned the settings, because default is to have a drop-down list of all existing users. With the default settings one would probably not notice anything different - because one never has to type in any user name and all user names are always "visible".

The how and the why is for devs to figure out that know / understand how these different packages interact with one another.

**Comment by CabrioletDiskette** [comment]
I was unable to replicate the issue. However, my best bet is that you have a conflicting line in /etc/lightdm/lightdm.conf. I would delete that file and all the lightdm config files in /etc/lightdm and then reinstall lightdm and all the lightdm greeters. After that add the greeter-hide-users=true line to /etc/lightdm/lightdm.conf again. If you have a complex lightdm configuration just temporarily back it up. Let me know what happens.

**Comment by gc-user** [comment]
I can replicate it every time, even if the conf file only contains that line and no other. Of course, the changes only get applied after reboot. And, of course, if this line is also removed, the bug can't be noticed as the user name list to choose from is displayed.

Also, I'm talking about 

/etc/lightdm/lightdm.conf.d/50-lightdm.conf,

not 

/etc/lightdm/lightdm.conf

The latter is the official / default file, the former is the one one can / should use for modifications.

But you wrote about lxdm. I couldn't determine if you tried to replicate the issue with lxdm being activated or not.
The bug seems to be happening between xfce4ui and one of the light* packages. Having lxdm installed might interfere with that and mask the bug.

Also, if that bug only appears after updating libxfce4ui to v4.18.4 and disappears after downgrading libxfce4ui to v4.18.3, why would uninstalling and reinstalling (the same versions of) lightdm / lightdm-gtk3-greeter have any effect?

Anyway, maybe one day, if there are no updates to the light* packages in the near future, or another future update to xfce4ui doesn't revert that bug, I might have to go and check the difference between the current and the previous version of xfce4ui to see if I maybe can detect something that could affect the login screen.

**Comment by CabrioletDiskette** [comment]
The testing I did was in a fresh install in a vm with only lightdm and xfce4. The idea behind the reinstall/remove exercise was to hope that the lightdm pkg replaces the configs with the default ones provided. Simply removing the pkg does not work as user created configs are not removed when a pkg is uninstalled. Try that out and see what happens. If that doesn't work than perhaps something else is at play that I need to replicate your problem. 

So in summary:

Super user(via sudo, su etc)

rm -r /etc/lightdm

xbps-remove -Rf lightdm lightdm-gtk3-greeter(or whatever one you have installed)

xbps-install -S lightdm lightdm-gtk3-greeter(whichever greeter you want)

Then add back the config lines you want.

Best of luck

**Comment by gc-user** [comment]
Thanks for your efforts.

Unfortunately, taking the folder away and letting the install process for the 3 mentioned light* packages create it anew with the respective default files didn't solve the problem on my system. Of course, I had to set

greeter-hide-users=true

in the respective conf file in order for the bug to be noticed. But, not surprisingly, it doesn't matter if the default file is edited or a new sub-folder and conf file is created for these kind of modified settings.

**Comment by CabrioletDiskette** [comment]
I have some news, maybe even good news. I was able to replicate your problem. Specifically if I put the greeter-hide-users tag above the [Seat*] header in the config file(any config file to be specific) the setting does not apply. Double check that you put greeter-hide-users under that specific tag in the .conf file. I would do it in the default one just for testings sake(it should not theoretically matter but just bear with me).

So your config should look like the one attached(see line 103)
[lightdm.conf.txt](https://github.com/void-linux/void-packages/files/11889151/lightdm.conf.txt)
d

If this does not work then I really got no more leads.

**Comment by gc-user** [comment]
I'm not sure you understand what symptom I described in the initial bug report.
Making a setting invalid by deliberately ignoring the correct placement of that setting in the respective file will make the corresponding service ignore that setting. This has neither anything to do with what I reported as symptom nor with the package which I reported being the seeming culprit when updating to the current version. That package most probably doesn't even read (or see or know about) the settings file of the other package.

**Comment by gc-user** [comment]
I just tested libxfce4ui v4.19.2 which shows the same issue. This tells me, that some other package, maybe LightDM needs adjusting to the new requirements of libxfce4ui - or they introduced a bug in v4.18.4 and it's still there (which is possible, but less probable).

**Comment by CabrioletDiskette** [comment]
I can't reproduce your problem so it likely a system issue not a bug(as I have said before). It is super hard to fix someone's system through comments on a GitHub issue.

I suggested to move the config line as you hinted that you may have made a "new sub-folder and conf file is created for these kind of modified settings" of you did not include the appropriate header in such a file then indeed the setting would not apply. In other words I am not saying this is a bug, but user error or some other system configuration problem. 

The other reason I doubt it is a bug is that like any display manager, lightdm interfaces with the .desktop files in /usr/share/xsessions/(note there is not really a intermediary packaged here as you suggested). Unless X got a major change(which it hasn't in 10years) then the only major way lightdm and xfce4 interface is via xsession.desktop files.

So best course of action is that someone else can reproduce your issue or you are able to fix your system. Keep troubleshooting, but I don't think I have any additional comments or help I can provide for this issue.
